src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -157,6 +163,81 @@ public boolean isPinned()
         return getTextChannel().clearReactionsById(getId());
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote, User user)
+    {
+        return getTextChannel().removeReactionById(getIdLong(), emote, user);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote, User user)
+    {
+        return getTextChannel().removeReactionById(getId(), emote, user);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(Emote emote)
+    {
+        Checks.notNull(emote, "Emote");
+
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().isEmote() && r.getReactionEmote().getEmote().equals(emote))
+            .findFirst().orElse(null);
+        return new ReactionPaginationActionImpl(reaction);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(String emote)
+    {
+        return retrieveReactionUsers(MessageReaction.ReactionEmote.fromUnicode(emote, getJDA()).getEmote());
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionByName(String name)
+    {
+        Checks.notNull(name, "Reaction name");
+
+        return this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.getName().equals(name))
+            .findFirst().orElse(null);
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionById(String id)
+    {
+        Checks.isSnowflake(id, "Reaction ID");
+
+        return this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.isEmote() && r.getIdLong() == Long.parseLong(id))
```

Comment:
Parse the long once and not for every emote

Reply:
Done